tests/utils/test_utils.go

```diff
@@ -0,0 +1,433 @@
+		LogInfoMessage(fmt.Sprintf("Parse and Validate %s : ", devfile.FileName))
+		devfile.ParsedSchemaObj, err = devfilepkg.ParseAndValidate(devfile.FileName)
+		if err != nil {
+			LogErrorMessage(fmt.Sprintf("From ParseAndValidate %v : ", err))
+		}
+		devfile.SchemaParsed = true
+	}
+	return err
+}
+
+// Verify verifies the contents of the specified devfile with the expected content
+func (devfile TestDevfile) Verify() error {
+
+	LogInfoMessage(fmt.Sprintf("Verify %s : ", devfile.FileName))
+
+	var errorString []string
+
+	err := devfile.parseSchema()
+
+	if err != nil {
+		errorString = append(errorString, LogErrorMessage(fmt.Sprintf("parsing schema %s : %v", devfile.FileName, err)))
+	} else {
+		LogInfoMessage(fmt.Sprintf("Get commands %s : ", devfile.FileName))
+		commands, _ := devfile.ParsedSchemaObj.Data.GetCommands(common.DevfileOptions{})
+		if commands != nil && len(commands) > 0 {
+			err = devfile.VerifyCommands(commands)
+			if err != nil {
+				errorString = append(errorString, LogErrorMessage(fmt.Sprintf("Verfify Commands %s : %v", devfile.FileName, err)))
+			}
+		} else {
+			LogInfoMessage(fmt.Sprintf("No command found in %s : ", devfile.FileName))
+		}
+	}
+
+	if err == nil {
+		LogInfoMessage(fmt.Sprintf("Get components %s : ", devfile.FileName))
+		components, _ := devfile.ParsedSchemaObj.Data.GetComponents(common.DevfileOptions{})
+		if components != nil && len(components) > 0 {
+			err = devfile.VerifyComponents(components)
+			if err != nil {
+				errorString = append(errorString, LogErrorMessage(fmt.Sprintf("Verfify Commands %s : %v", devfile.FileName, err)))
+			}
+		} else {
+			LogInfoMessage(fmt.Sprintf("No components found in %s : ", devfile.FileName))
+		}
+	}
+
+	var returnError error
+	if len(errorString) > 0 {
+		returnError = errors.New(fmt.Sprint(errorString))
+	}
+	return returnError
+
+}
+
+// EditCommands modifies random attributes for each of the commands in the devfile.
+func (devfile TestDevfile) EditCommands() error {
+
+	LogInfoMessage(fmt.Sprintf("Edit %s : ", devfile.FileName))
+
+	err := devfile.parseSchema()
+	if err != nil {
+		LogErrorMessage(fmt.Sprintf("From parser : %v", err))
+	} else {
+		LogInfoMessage(fmt.Sprintf(" -> Get commands %s : ", devfile.FileName))
+		commands, _ := devfile.ParsedSchemaObj.Data.GetCommands(common.DevfileOptions{})
+		for _, command := range commands {
+			err = devfile.UpdateCommand(&command)
+			if err != nil {
+				LogErrorMessage(fmt.Sprintf("Updating command : %v", err))
+			} else {
+				LogInfoMessage(fmt.Sprintf("Update command in Parser : %s", command.Id))
+				devfile.ParsedSchemaObj.Data.UpdateCommand(command)
+			}
+		}
+		LogInfoMessage(fmt.Sprintf("Write updated file to yaml : %s", devfile.FileName))
+		devfile.ParsedSchemaObj.WriteYamlDevfile()
+		devfile.SchemaParsed = false
+	}
+	return err
+}
+
+// EditComponents modifies random attributes for each of the components in the devfile.
+func (devfile TestDevfile) EditComponents() error {
```

Comment:
should this be in component utils file?

Reply:
I prefer where it is because the use the function makes of the parser, wanted to keep the component_utils pure component stuff.